src/main/java/slimeknights/tconstruct/common/data/TConstructRecipeProvider.java

```diff
@@ -1079,6 +1160,92 @@ private void addSearedStonecutter(@Nonnull Consumer<IFinishedRecipe> consumer, I
+      .setCast(TinkerSmeltery.ingotCast, false)
+      .addCriterion("has_item", this.hasItem(ingot))
+      .build(consumer, location("casting/" + ingot.asItem().getRegistryName().getPath()));
+  }
+
+  private void addNuggetCastingRecipe(@Nonnull Consumer<IFinishedRecipe> consumer, FluidObject fluidObject, IItemProvider nugget, int amount) {
+    CastingRecipeBuilder.tableRecipe(nugget)
+      .setFluid(new FluidStack(fluidObject.get(), 16))
+      .setCast(TinkerSmeltery.nuggetCast, false)
+      .addCriterion("has_item", this.hasItem(nugget))
+      .build(consumer, location("casting/" + nugget.asItem().getRegistryName().getPath()));
+  }
+
+  /**
+   * Adds slime related casting recipes
+   * @param consumer    Recipe consumer
+   * @param fluidObject Fluid matching the slime type
+   * @param slimeType   SlimeType for this recipe
+   */
+  private void addSlimeCastingRecipe(@Nonnull Consumer<IFinishedRecipe> consumer, @Nonnull FluidObject fluidObject, SlimeType slimeType) {
+    addBlockCastingRecipe(consumer, fluidObject, TinkerWorld.congealedSlime.get(slimeType), MaterialValues.VALUE_BrickBlock);
+    CastingRecipeBuilder.basinRecipe(TinkerWorld.slime.get(slimeType))
+      .setFluid(new FluidStack(fluidObject.get(), 720))
+      .setCast(TinkerWorld.congealedSlime.get(slimeType), true)
+      .addCriterion("has_item", this.hasItem(TinkerCommons.slimeball.get(slimeType)))
+      .build(consumer, location("casting/slime/" + slimeType.getName()));
+    CastingRecipeBuilder.tableRecipe(TinkerCommons.slimeball.get(slimeType))
+      .setFluid(new FluidStack(fluidObject.get(), MaterialValues.VALUE_Ingot))
+      .addCriterion("has_item", this.hasItem(TinkerCommons.slimeball.get(slimeType)))
+      .build(consumer, location("casting/slimeball/" + slimeType.getName()));
+    this.addBucketCastingRecipe(consumer, fluidObject);
+  }
+
+  private void addMoltenMineralCastingRecipe(@Nonnull Consumer<IFinishedRecipe> consumer, @Nonnull FluidObject fluidObject, @Nullable IItemProvider block, @Nullable IItemProvider ingot, @Nullable IItemProvider nugget) {
+    addBucketCastingRecipe(consumer, fluidObject);
+    if (block != null) {
+      addBlockCastingRecipe(consumer, fluidObject, block, MaterialValues.VALUE_Block);
+    }
+    if (ingot != null) {
+      addIngotCastingRecipe(consumer, fluidObject, ingot, MaterialValues.VALUE_Ingot);
+    }
+    if (nugget != null) {
+      addNuggetCastingRecipe(consumer, fluidObject, nugget, MaterialValues.VALUE_Nugget);
+    }
+  }
+
+  private void addCastCastingRecipe(@Nonnull Consumer<IFinishedRecipe> consumer, @Nonnull Tag<Item> tag, @Nonnull IItemProvider cast) {
+    CastingRecipeBuilder.tableRecipe(cast)
+      .setFluid(new FluidStack(TinkerFluids.moltenGold.get(), MaterialValues.VALUE_Ingot))
+      .setCast(tag, true)
+      .setSwitchSlots()
+      .addCriterion("has_item", this.hasItem(tag))
+      .build(consumer, location("casting/casts/" + tag.getId().getPath()));
```

Comment:
Typically, we name recipes based on their output.

Reply:
Is this not? `tconstruct:casting/casts/nuggets` is a cast for nuggets?

Reply:
The tag ID may be similar to the output, but its not strictly the same. this is more different for the item version than the tag one, where the input may be a specific material like bricks for the ingot cast.